qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,246 @@
+	private AppointmentBlocksPage appointmentBlocksPage;
+	private ManageAppointmentsPage manageAppointmentsPage;
+	private ManageProviderSchedulesPage manageProviderSchedulesPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+	private static final String FIRST_SERVICE_NAME = "Oncology";
+	private static final String SECOND_SERVICE_NAME = "Dermatology";
+	private static final String LOCATION = "Laboratory";
+
+	@Before(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void visitDashboard() {
+		initiateWithLogin();
+		testPatient = createTestPatient();
+		findPatientPage = homePage.goToFindPatientRecord();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Request Appointment link from the Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage.clickOnRequest();
+	}
+
+	@When("the system loads Request Appointment page")
+	public void systemLoadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@And("a user fills in the appointment request details")
+	public void captureAppointmentRequestDetails() {
+		requestAppointmentPage.enterAppointmentType(FIRST_SERVICE_NAME);
+		requestAppointmentPage.enterMinimumValue("0");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void userClicksOnSaveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
+	}
+
+	@When("a user clicks on Schedule Appointment link from the Patient dashboard")
+	public void loadManageAppointmentsPage() {
+		manageAppointmentsPage = (ManageAppointmentsPage) dashboardPage.goToManageAppointments();
+	}
+
+	@And("the system loads Manage Appointments page")
+	public void systemLoadsManageAppointmentsPage() {
+		assertTrue(manageAppointmentsPage.containsText(testPatient.identifier));
+	}
+
+	@And("a user clicks on Cancel this Request icon")
+	public void userClicksOnDeleteAppointmentRequest() {
+		manageAppointmentsPage.deleteRequest() ;
+	}
+
+	@Then("the system deletes appointment request from the request table")
+	public void systemDeletesAppointmentRequest() {
+		assertTrue(manageAppointmentsPage.containsText("No appointment requests"));
+	}
+
+	@When("a user clicks Appointment scheduling app from the home page")
```

Comment:
```suggestion
	@When("a user clicks  on Appointment scheduling app from the home page")
```

Reply:
@sherrif10 If I am to implement this change, the test will not run because the proposed change already exists in the VitalsAndTriagingSteps.